walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -1,23 +1,100 @@
-// Copyright 2019 Google LLC
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     https://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package com.google.sps;
-
-import java.util.Collection;
-
-public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
-}
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import java.sql.Time;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+public final class FindMeetingQuery {
+  public static final Collection<TimeRange> EMPTY_QUERY = new ArrayList<>();
+  public static final Collection<TimeRange> FULL_QUERY = Stream.of(TimeRange.WHOLE_DAY).collect(Collectors.toList());
+  /**
+   * Assumptions: request and events are all valid, but can be non-null.
+   * - Request has duration larger than 0, and at least one attendee.
+   * - Each Event has a time range within the day, and may have no attendees.
+   */
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+    if (events == null || request == null) {
+      throw new RuntimeException("input variables should be non-null!");

```

Comment:
`RuntimeException` is not accurate here, if no newly defined check exception is used, consider using `IllegalArgumentException` instead.